app/src/org/commcare/views/media/CustomVideoView.java

```diff
@@ -0,0 +1,49 @@
+package org.commcare.views.media;
+
+import android.content.Context;
+import android.widget.VideoView;
+import java.util.Date;
+
+/**
+ * A custom VideoView which records the total video play duration
+ * and send the update via {@link org.commcare.views.media.CustomVideoView.TimeSpentListener#getPlayDuration(long) getPlayDuration}
+ * @author $|-|!˅@M
+ */
+public class CustomVideoView extends VideoView {
+
+    private TimeSpentListener listener;
+
+    private long duration;
+    private long startTime;
+
+    public CustomVideoView(Context context) {
+        super(context);
+    }
+
+    public void setListener(TimeSpentListener listener) {
+        this.listener = listener;
+    }
+
+    @Override
+    public void pause() {
+        super.pause();
+        duration += new Date().getTime() - startTime;
+    }
+
+    @Override
+    public void start() {
+        super.start();
+        startTime = new Date().getTime();
+    }
+
+    @Override
+    protected void onDetachedFromWindow() {
+        super.onDetachedFromWindow();
+        listener.getPlayDuration(duration);
+    }
+
+    public interface TimeSpentListener {
+        void getPlayDuration(long duration);
```

Comment:
Since it's a listener I would like to call it "OnVideoDetached" or something similar.

Reply:
@ShivamPokhriyal  looks like this is not resolved yet.

Reply:
I made a force-push maybe that's why it's not showing resolved here. But you can check the code [here](https://github.com/dimagi/commcare-android/pull/2166/files#diff-1a0cb3908dcc92f87e97b0f9d670d968R45)

Reply:
I still see the method called `getPlayDuration` on your link and even [here](https://github.com/dimagi/commcare-android/blob/inline-video-analytics/app/src/org/commcare/views/media/CustomVideoView.java#L42)

Also the PR only has a single commit. So don't think it was pushed.